google_guest_agent/telemetry/telemetry_test.go

```diff
@@ -0,0 +1,65 @@
+//  Copyright 2023 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package telemetry
+
+import (
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+func TestRecordTelemetry(t *testing.T) {
+	var got http.Header
+	handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Can you start a "fake" implementation in the metadata package in way that you can use it here? You don't need to make it "perfect", just make sure it's centralized with the metadata package so we can start improving/increasing its coverage in a single place.